src/compile-middleware.ts

```diff
@@ -46,47 +46,51 @@ function isSuccessful(response: Response) {
   return (statusCode >= 200 && statusCode < 300);
 }
 
-export const babelCompile: RequestHandler = transformResponse({
-  shouldTransform(_request: Request, response: Response) {
-    return isSuccessful(response) &&
-        compileMimeTypes.indexOf(getContentType(response)) >= 0;
-  },
-
-  transform(request: Request, response: Response, body: string): string{
-    const contentType = getContentType(response);
-    const uaParser = new UAParser(request.headers['user-agent']);
-    const compile = needCompilation(uaParser);
-
-    if (compile) {
-      // TODO(justinfagnani): cache compilation results
-      if (contentType === htmlMimeType) {
-        const document = parse5.parse(body);
-        const scriptTags = dom5.queryAll(document, isInlineJavaScript);
-        for (const scriptTag of scriptTags) {
+export function babelCompile(forceCompile: boolean): RequestHandler {
+  if (forceCompile == null)
+    forceCompile = false;
```

Comment:
done. I had written this as a single-line if, which I'm in favor of, so we should also see if clang can let those through.